### ISSUE TITLE
Port drag updates to the frontend

### ISSUE THREAD

**Issue by wodeni** [penrose/penrose#242]
The new server does not handle `dragUpdate` now. We ported basic drag events to the frontend (the ones changing `x` and `y`), but still need to port the remaining logic to the frontend.

```haskell
dragShape shape dx dy
    | shape `is` "Line" =
        trRaw "here"  $ move "startX" dx $
        move "startY" dy $
        move "endX"   dx $
        move "endY"   dy shape
    | shape `is` "Image" =
        move "centerX" dx $
        move "centerY" dy shape
    | shape `is` "Arrow" =
        move "startX" dx $
        move "startY" dy $
        move "endX"   dx $
        move "endY"   dy shape
    | shape `is` "Curve" =
      movePath (dx, dy) shape
    | otherwise = setX (FloatV (dx + getX shape)) $ setY (FloatV (dy + getY shape)) shape
```

**Comment by maxkrieger** [comment]
Is this a thing still? For arrow endpoints I'd imagine?

**Comment by wodeni** [comment]
Yeah, for any shape without primary (x, y) coords.

**Comment by wodeni** [comment]
Resolved in #406
